server/src/main/kotlin/fi/oph/kitu/yki/YkiViewController.kt

```diff
@@ -14,27 +14,31 @@ class YkiViewController(
 ) {
     @GetMapping("/suoritukset", produces = ["text/html"])
     fun suorituksetView(
-        @RequestParam("versionHistory") versionHistory: Boolean?,
-        @RequestParam("limit") limit: Int = 100,
-        @RequestParam("page") page: Int = 1,
+        search: String = "",
+        versionHistory: Boolean = false,
+        limit: Int = 100,
+        page: Int = 1,
     ): ModelAndView {
-        val suorituksetTotal = ykiService.countSuoritukset(versionHistory)
+        val suorituksetTotal = ykiService.countSuoritukset(search, versionHistory)
         val totalPages = ceil(suorituksetTotal.toDouble() / limit).toInt()
         val offset = limit * (page - 1)
         val nextPage = if (page >= totalPages) null else page + 1
         val previousPage = if (page <= 1) null else page - 1
+        val searchStrUrl = URLEncoder.encode(search, Charsets.UTF_8)
         val paging =
             mapOf(
                 "totalEntries" to suorituksetTotal,
                 "currentPage" to page,
                 "nextPage" to nextPage,
                 "previousPage" to previousPage,
                 "totalPages" to totalPages,
+                "searchStr" to search,
+                "searchStrUrl" to searchStrUrl,
             )
```

Comment:
Tätä sivutuspalikan konstruointia alkaa olla sen verran, että voisi olla siistimpää tehdä se omana funktionaan ja/tai luoda sille oma dataluokka. Nuo laskutoimitukset yms. saattaisi solahtaa aika luontevasti konstruktoriin. Ehkä.

Kuvittelisin myös, että tuon `mapOf`-osuuden saisi myös melko helposti johdettua luokan kentistä. Tuollainen alati kasvava string-to-object -mappaus aiheuttaa javadinossa hieman jäsä-ihottumaa.

Ei mitään todellisia ongelmia tässä siis, mutta lähinnä ideoiden pallottelua, miten näitä saataisiin tehtyä mahdollisimman siististi jatkossa. Tää on varmasti ihan tarpeeksi siisti nytkin.